src/fread.c

```diff
@@ -2035,7 +2035,10 @@ int freadMain(freadMainArgs _args) {
     }
     if (verbose) {
       if (sampleLines==0) {
-        DTPRINT(_("  'header' determined to be %s because there are%s number fields in the first and only row\n"), args.header?"TRUE":"FALSE", args.header?_(" no"):"");
+        if (args.header)
+          DTPRINT(_("  'header' determined to be TRUE because there are no number fields in the first and only row\n"));
+        else
+          DTPRINT(_("  'header' determined to be FALSE because there are number fields in the first and only row\n"));
```

Comment:
Curious to know if the extra space before 'header' is required for something

Reply:
Not "required", but it's consistent; `verbose=TRUE` output uses "heading/indented" to organize the large amount of output:

```
1301:    if (verbose) DTPRINT(_("  Using %d threads (omp_get_max_threads()=%d, nth=%d)\n"), nth, maxth, args.nth);
1341:      DTPRINT(_("  No NAstrings provided.\n"));
1343:      DTPRINT(_("  NAstrings = ["));
1348:        DTPRINT(_("  One or more of the NAstrings looks like a number.\n"));
1350:        DTPRINT(_("  None of the NAstrings look like numbers.\n"));
1352:    if (args.skipNrow >= 0) DTPRINT(_("  skip num lines = %"PRId64"\n"), (int64_t)args.skipNrow);
1353:    if (args.skipString) DTPRINT(_("  skip to string = <<%s>>\n"), args.skipString);
1354:    DTPRINT(_("  show progress = %d\n"), args.showProgress);
1355:    DTPRINT(_("  0/1 column will be read as %s\n"), args.logical01? "boolean" : "integer");
1389:    if (verbose) DTPRINT(_("  `input` argument is provided rather than a file name, interpreting as raw text to read\n"));
1396:    if (verbose) DTPRINT(_("  Opening file %s\n"), args.filename);
1408:      if (verbose) DTPRINT(_("  File opened, size = %s.\n"), filesize_to_str(fileSize));
1440:      if (verbose) DTPRINT(_("  File opened, size = %s.\n"), filesize_to_str(fileSize));
1453:    if (verbose) DTPRINT(_("  Memory mapped ok\n"));
1476:    if (verbose) DTPRINT(_("  UTF-8 byte order mark EF BB BF found at the start of the file and skipped.\n"));
1489:    if (verbose) DTPRINT(_("  Last byte(s) of input found to be %s and removed.\n"),
1506:    _("  No \\n exists in the file at all, so single \\r (if any) will be taken as one line ending. This is unusual but will happen normally when there is no \\r either; e.g. a single line missing its end of line.\n") :
1507:    _("  \\n has been found in the input and different lines can end with different line endings (e.g. mixed \\n and \\r\\n in one file). This is common and ideal.\n"));
1531:        if (verbose) DTPRINT(_("  File ends abruptly with '%c'. Final end-of-line is missing. Using cow page to write 0 to the last byte.\n"), eof[-1]);
1539:          DTPRINT(_("  File ends abruptly with '%c'. Final end-of-line is missing. Copying file in RAM. %s.\n"), eof[-1], msg);
1548:          DTPRINT(_("  File copy in RAM took %.3f seconds.\n"), time_taken);
1608:      DTPRINT(_("  Skipped to line %"PRIu64" in the file"), (uint64_t)row1line);
1624:    if (lineStart>ch) DTPRINT(_("  Moved forward to first non-blank line (%d)\n"), row1line);
1625:    DTPRINT(_("  Positioned on line %d starting: <<%s>>\n"), row1line, strlim(lineStart, 30));
1651:    if (verbose) DTPRINT(_("  sep='\\n' passed in meaning read lines as single character column\n"));
1670:      if (verbose) DTPRINT(_("  Detecting sep automatically ...\n"));
1677:      if (verbose) DTPRINT(_("  Using supplied sep '%s'\n"), args.sep=='\t' ? "\\t" : seps);
1692:        // if (verbose) DTPRINT(_("  Trying sep='%c' with quoteRule %d ...\n"), sep, quoteRule);
1710:              DTPRINT((unsigned)sep<32 ? "  sep=%#02x" : "  sep='%c'", sep);
1711:              DTPRINT(_("  with %d fields using quote rule %d\n"), topNumFields, quoteRule);
1760:              DTPRINT((unsigned)sep<32 ? "  sep=%#02x" : "  sep='%c'", sep);
1761:              DTPRINT(_("  with %d lines of %d fields using quote rule %d\n"), topNumLines, topNumFields, topQuoteRule);
1768:      if (verbose) DTPRINT(_("  No sep and quote rule found a block of 2x2 or greater. Single column input.\n"));
1810:    DTPRINT(_("  Detected %d columns on line %d. This line is either column names or first data row. Line starts as: <<%s>>\n"),
1812:    DTPRINT(_("  Quote rule picked = %d\n"), quoteRule);
1813:    DTPRINT(_("  fill=%s and the most number of columns found is %d\n"), fill?"true":"false", ncol);
1821:        DTPRINT(_("  Copying file in RAM. %s\n"), msg);
1830:        DTPRINT(_("  File copy in RAM took %.3f seconds.\n"), time_taken);
1836:      if (verbose) DTPRINT(_("  1-column file ends with 2 or more end-of-line. Restoring last eol using extra byte in cow page.\n"));
1856:  if (verbose && args.header!=NA_BOOL8) DTPRINT(_("  'header' changed by user from 'auto' to %s\n"), args.header?"true":"false");
1867:      DTPRINT(_("  sep=',' so dec set to '.'\n"));
1891:    DTPRINT(_("  Number of sampling jump points = %d because "), nJumps);
1933:        if (verbose) DTPRINT(_("  A line with too-%s fields (%d/%d) was found on line %d of sample jump %d. %s\n"),
1957:        DTPRINT(_("  dec='%c' detected based on a balance of %d parsed fields\n"), dec, abs(linesForDecDot));
1967:      DTPRINT(_("  Type codes (jump %03d)    : %s  Quote rule %d\n"), jump, typesAsString(ncol), quoteRule);
1983:        DTPRINT(_("  dec='%c' detected based on a balance of %d parsed fields\n"), dec, abs(linesForDecDot));
1989:        if (verbose) DTPRINT(_("  'header' determined to be true due to column %d containing a string on row 1 and a lower type (%s) in the rest of the %"PRId64" sample rows\n"),
2038:        DTPRINT(_("  'header' determined to be %s because there are%s number fields in the first and only row\n"), args.header?"TRUE":"FALSE", args.header?_(" no"):"");
2041:          DTPRINT(_("  'header' determined to be true because all columns are type string and a better guess is not possible\n"));
2043:          DTPRINT(_("  'header' determined to be false because there are some number columns and those columns do not have a string field at the top of them\n"));
2059:    if (verbose && bumped) DTPRINT(_("  Type codes (first row)   : %s  Quote rule %d\n"), typesAsString(ncol), quoteRule);
2068:    if (verbose) DTPRINT(_("  All rows were sampled since file is small so we know nrow=%"PRIu64" exactly\n"), (uint64_t)sampleLines);
2080:      DTPRINT(_("  =====\n"));
2081:      DTPRINT(_("  Sampled %"PRIu64" rows (handled \\n inside quoted fields) at %d jump points\n"), (uint64_t)sampleLines, nJumps);
2082:      DTPRINT(_("  Bytes from first data row on line %d to the end of last row: %"PRIu64"\n"), row1line, (uint64_t)bytesRead);
2083:      DTPRINT(_("  Line length: mean=%.2f sd=%.2f min=%d max=%d\n"), meanLineLen, sd, minLen, maxLen);
2084:      DTPRINT(_("  Estimated number of rows: %"PRIu64" / %.2f = %"PRIu64"\n"), (uint64_t)bytesRead, meanLineLen, (uint64_t)estnrow);
2085:      DTPRINT(_("  Initial alloc = %"PRIu64" rows (%"PRIu64" + %d%%) using bytes/max(mean-2*sd,min) clamped between [1.1*estn, 2.0*estn]\n"),
2087:      DTPRINT(_("  =====\n"));
2093:    if (verbose) DTPRINT(_("  Alloc limited to lower nrows=%"PRIu64" passed in.\n"), (uint64_t)nrowLimit);
2158:    if (verbose) DTPRINT(_("  Cancelled by user: userOverride() returned false.")); // # nocov
2191:  if (verbose) DTPRINT(_("  After %d type and %d drop user overrides : %s\n"), nUserBumped, ndrop, typesAsString(ncol));
2200:    DTPRINT(_("  Allocating %d column slots (%d - %d dropped) with %"PRIu64" rows\n"),
2261:  if (verbose) DTPRINT(_("  jumps=[%d..%d), chunk_size=%"PRIu64", total_size=%"PRIu64"\n"),
2605:      DTPRINT(_("  Provided number of fill columns: %d but only found %d\n"), ncol, max_col);
2606:      DTPRINT(_("  Dropping %d overallocated columns\n"), ndropFill);
2631:      if (verbose) DTPRINT(_("  Too few rows allocated. Allocating additional %"PRIu64" rows (now nrows=%"PRIu64") and continue reading from jump %d\n"),
2638:      if (verbose) DTPRINT(_("  Restarting team from jump %d. nSwept==%d quoteRule==%d\n"), jump0, nSwept, quoteRule);
2658:      if (verbose) DTPRINT(_("  %d out-of-sample type bumps: %s\n"), nTypeBump, typesAsString(ncol));
2701:    DTPRINT(_("  Type counts:\n"));
```

Reply:
Noted!